lib/mongo/index/view.rb

```diff
@@ -251,6 +254,10 @@ def send_initial_query(server)
 
       def with_generated_names(models)
         models.dup.each do |model|
+          if model[:key].key?(:partial_filter_expression)
+            model[:partial_filter_expression] = model[:key][:partial_filter_expression]
+            model[:key].delete(:partial_filter_expression)
+          end
```

Comment:
I have to double check, but I don't believe this code is necessary. Since you added this line https://github.com/mongodb/mongo-ruby-driver/pull/697/files#diff-4104300dfb8ef7232ede5d88a9008676R59
it should probably set the partial filter expression option correctly.